p2p/pre2p/network.go

```diff
@@ -0,0 +1,98 @@
+package pre2p
+
+import (
+	"fmt"
+	"log"
+	"net"
+
+	"github.com/pokt-network/pocket/p2p/pre2p/types"
+	pcrypto "github.com/pokt-network/pocket/shared/crypto"
+)
+
+const (
+	NetworkProtocol = "tcp4"
+)
+
+var _ types.Network = &network{}
+
+type network struct {
+	// TODO(team): This address book is currently static and does not update dynamically as new peers come on/offline.
+	AddrBook []*types.NetworkPeer
+}
+
+func ConnectToValidatorNetwork(validators types.ValMap) (n types.Network) {
+	n = &network{}
+	for _, v := range validators {
+		err := n.(*network).connectToValidator(v)
+		if err != nil {
+			log.Println("[WARN] Error connecting to validator: ", err)
+			continue
+		}
+	}
+	return
+}
+
+// TODO(olshansky): How do we avoid self-broadcasts?
+func (n *network) NetworkBroadcast(data []byte) error {
+	for _, peer := range n.AddrBook {
+		client, err := net.DialTCP(NetworkProtocol, nil, peer.ConsensusAddr)
+		if err != nil {
+			log.Println("Error connecting to one of the peers during broadcast: ", err)
+			continue
+		}
+		defer client.Close()
+
+		_, err = client.Write(data)
+		if err != nil {
+			log.Println("Error writing to one of the peers during broadcast: ", err)
+			continue
+		}
+	}
+	return nil
+}
+
+func (n *network) NetworkSend(data []byte, node pcrypto.Address) error {
+	for _, peer := range n.AddrBook {
+		// TODO(team): If the address book is a map instead of a list, we wouldn't have to do this loop.
```

Comment:
This contains the answer to the question at [line 35](https://github.com/pokt-network/pocket/pull/33/files#diff-4c195a5d75f98cf6df9e78d6fc0766ada49a8a40af22a3c4385d3c4becc008dcR35) as a list containing only neighbors would be a self-filtering mechanism to prevent self-broadcast, as long as other peers do not re-broadcast to not fall into infinite broadcast.

Reply:
I suppose what I implied since Pre2P is always going to be used in controlled environments that we establish the expectation that the neighbors list will never contain own address, making of it a natural self-filtering mechanism. That's why making it a list and not a map will be much more useful.

Reply:
Makes sense. We'll just need to validate that when the static list / addrbook is created /configured, self is not in there.

Added a comment in the code.